sdk-lib/src/main/java/cash/z/ecc/android/sdk/SdkSynchronizer.kt

```diff
@@ -661,65 +644,38 @@ class SdkSynchronizer internal constructor(
     override suspend fun getLegacyTransparentAddress(account: Account): String =
         processor.getTransparentAddress(account)
 
-    override fun sendToAddress(
+    override suspend fun sendToAddress(
         usk: UnifiedSpendingKey,
         amount: Zatoshi,
         toAddress: String,
         memo: String
-    ): Flow<PendingTransaction> = flow {
-        twig("Initializing pending transaction")
+    ): Flow<PendingTransaction> {
         // Emit the placeholder transaction, then switch to monitoring the database
-        txManager.initSpend(amount, toAddress, memo, usk.account).let { placeHolderTx ->
-            emit(placeHolderTx)
-            txManager.encode(usk, placeHolderTx).let { encodedTx ->
-                // only submit if it wasn't cancelled. Otherwise cleanup, immediately for best UX.
-                if (encodedTx.isCancelled()) {
-                    twig("[cleanup] this tx has been cancelled so we will cleanup instead of submitting")
-                    if (cleanupCancelledTx(encodedTx)) {
-                        refreshAllBalances()
-                    }
-                } else {
-                    txManager.submit(encodedTx)
-                }
-            }
+        val placeHolderTx = txManager.initSpend(amount, toAddress, memo, usk.account)
+
+        txManager.encode(usk, placeHolderTx).let { encodedTx ->
+            txManager.submit(encodedTx)
         }
-    }.flatMapLatest {
-        // switch this flow over to monitoring the database for transactions
-        // so we emit the placeholder TX above, then watch the database for all further updates
-        twig("Monitoring pending transaction (id: ${it.id}) for updates...")
-        txManager.monitorById(it.id)
-    }.distinctUntilChanged()
-
-    override fun shieldFunds(
+
+        return txManager.monitorById(placeHolderTx.id)
+    }
+
+    override suspend fun shieldFunds(
         usk: UnifiedSpendingKey,
         memo: String
-    ): Flow<PendingTransaction> = flow {
+    ): Flow<PendingTransaction> {
         twig("Initializing shielding transaction")
-        // TODO(str4d): This only shields funds from the current UA's transparent receiver. Fix this once we start
-        //  rolling UAs.
```

Comment:
This TODO still applies. We should ensure there is an issue tracking this if the code is not.